src/external/atomic_queue/defs.h

```diff
@@ -6,7 +6,8 @@
 
 #include <atomic>
 
-#if defined(__x86_64__) || defined(_M_X64)
+#if defined(__x86_64__) || defined(_M_X64) || \
+    defined(__i386__) || defined(_M_IX86)
```

Comment:
As with ARM 32bits I am not sure whether this restriction was intentional or just untested in the original `atomic_queue`, for ARM it seemed to work ok...

Reply:
The intrinsic _mm_pause is part of SSE2. I didn't see a reason why it should be disabled.
It's also why to enable "-msse2" in gcc flags, as it's not part of defaults on 32bit.

I wonder, if one wants a generic `spin_loop_pause` for all platforms, can we fallback to OS syscalls? (cf. `pthread_yield`, `sched_yield`) Then, I can enable more platforms to be compatible, like powerpc.

Reply:
There's also (https://en.cppreference.com/w/cpp/thread/yield) but I don't know exactly how all of these differ. Most conversations I've seen tend to suggest not to assume too much about the behavior of these calls across platforms.

Reply:
When you use syscall, rather than instructions, you are hoping that the OS will handle this in a RT-compliant manner, which may or may not be, and it's a high chance that this call does this.

Also I find we can do as in boost, which implements for various cpus:
https://www.boost.org/doc/libs/1_64_0/boost/fiber/detail/cpu_relax.hpp

Reply:
Heh the boost solution looks nice. This "wait 0us" is a bit strange though, this looks like an assumption on how the scheduler will handle the particular case of a zero-duration sleep. At this point using `std::this_thread::yield()` directly has the correct semantic at least.

Reply:
Ok, I agree totally, I think to add it as a following PR, not this.
This one is already loaded as it is, + it has the 32bit which is not strictly a thing of MinGW.